Src/IronPython/Runtime/LiteralParser.cs

```diff
@@ -42,15 +53,20 @@ public static class LiteralParser {
                             throw PythonOps.ValueError("Trailing \\ in string");
                         }
                     }
-                    ch = text[i++];
+                    ch = Convert.ToChar(data[i++]);
 
                     if (ch == 'u' || ch == 'U') {
                         int len = (ch == 'u') ? 4 : 8;
                         int max = 16;
                         if (isUniEscape) {
-                            if (TryParseInt(text, i, len, max, out val)) {
+                            if (TryParseInt(data, i, len, max, out val, out int consumed)) {
                                 if (val < 0 || val > 0x10ffff) {
-                                    throw PythonExceptions.CreateThrowable(PythonExceptions.UnicodeDecodeError, isRaw ? "rawunicodeescape" : "unicodeescape", Bytes.Empty, i - start - 2, i - start + len - 1, "illegal Unicode character");
+                                    throw PythonExceptions.CreateThrowable(
+                                        PythonExceptions.UnicodeDecodeError,
+                                        isRaw ? "rawunicodeescape" : "unicodeescape",
+                                        byteData != null ? new Bytes(byteData) : Bytes.Empty,
```

Comment:
Maybe `data is bytes[] byteData ? new Bytes(byteData) : Bytes.Empty` and avoid having to pass in an extra argument? Although if we end up needing two overloads it's a useless suggestions.

Reply:
Agreed. I have added the extra argument in case a (future) user of the `char[]` overload wanted to provide the byte data for the sake of having a coplete `UnicodeDecodeError`, but looking at the usage it is unlikely ever needed - the exception is caught and recast as a syntax error. And it is always possible to catch-modify-rethow if needed, just like `StringOps.DoDecode` does.

I still hope we will make the generics work somehow. There is already plenty of very similar parsing code to maintain. We'll see.